libzeth/circuits/blake2s/blake2s_comp.hpp

```diff
@@ -45,28 +41,89 @@ class BLAKE2s_256_comp : public libsnark::gadget<FieldT>
     // We use the workaround described here
     // https://stackoverflow.com/questions/32912921/whats-wrong-with-this-inline-initialization-of-stdarray
     // to initialize the const std::arrays
-    const std::array<bool, BLAKE2s_word_size> f0 = {{
+    const std::array<bool, BLAKE2s_word_size> flag_to_1 = {{
         1, 1, 1, 1, 1, 1, 1, 1, // FF
         1, 1, 1, 1, 1, 1, 1, 1, // FF
         1, 1, 1, 1, 1, 1, 1, 1, // FF
         1, 1, 1, 1, 1, 1, 1, 1  // FF
     }};
 
-    // We use the sequential mode, f1 is set to x00000000
-    const std::array<bool, BLAKE2s_word_size> f1 = {{
-        0, 0, 0, 0, 0, 0, 0, 0, // 00
-        0, 0, 0, 0, 0, 0, 0, 0, // 00
-        0, 0, 0, 0, 0, 0, 0, 0, // 00
-        0, 0, 0, 0, 0, 0, 0, 0  // 00
-    }};
+    // See: Appendix A.1 of https://blake2.net/blake2.pdf for the specification
+    // of the permutations used in BLAKE2s
+    std::array<std::array<uint, 16>, 10> sigma = {
```

Comment:
These are only permutation constants, they are independent of the size word (blake2b uses the same constants while it manipulates 64 bit word).

Reply:
Oh yes ofc.. let's switch `uint` to `uint8_t` since the sigma matrix contains the permutations of the elements of the set {0, ..., 15} which binary representations fit in 8 bits

Reply:
Sure ! Updated.